### PR TITLE
chore: refactor contributing guidelines for humans and agents

### PR DESCRIPTION
**Pull request by rclement**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+
+Follow the following pragmatic software engineering practices:
+
+- Avoid over-engineered solutions, prefer simplicity and add complexity only when required (start small and simple)
+- Avoid premature abstractions, especially in the beginning (e.g. prefer having a single well defined module than ten different ones containing a single entity each)
+- Prefer explicit symbol names and logic structure over extensive comments
+- Prefer functional programming over object-oriented programming (use the "functional core, imperative shell" pattern)
+- When writing tests, use only test functions with well-thought fixture dependencies, but avoid test classes
+- Test suites should match the source module architecture (e.g. `my_module.py` should have a `test_my_module.py` test suite)
+- Design tests in a top-bottom fashion: start with functional tests from the user perspective with maximum dependencies, then focus with unit testing on chirurgical parts when needed to test subtle edge-cases
```
[the remark this answers is not in the document]
Comment (rclement) [reply CONTRIBUTING.md]: Done

File: CLAUDE.md [hunk 2]
```diff
@@ -1,43 +1 @@
...
-- Type checking uses strict mypy
-
-## Releases
-
-- Version tags have **no `v` prefix** (e.g. `0.1.0`, not `v0.1.0`)
-- Version is defined in `pyproject.toml` under `[project] version`
-- Changelog follows [Keep a Changelog](https://keepachangelog.com/) format in `CHANGELOG.md`
-- Release branches are named `release/<version>`
-- Release commit message: `chore: release version <version>`
+@CONTRIBUTING.md
```
[the remark this answers is not in the document]
Comment (rclement) [reply CLAUDE.md]: Claude Code uses this syntax
